src/Models/HydrostaticFreeSurfaceModels/mg_implicit_free_surface_solver.jl

```diff
@@ -56,10 +60,13 @@ step `Δt`, gravitational acceleration `g`, and free surface at the `n`-th time-
 """
 function MGImplicitFreeSurfaceSolver(grid::AbstractGrid, 
                                      settings = nothing,
-                                     gravitational_acceleration = g_Earth, 
+                                     gravitational_acceleration = g_Earth,
+                                     reduced_dim = (false, false, true),
                                      placeholder_timestep = -1.0)
     arch = architecture(grid)
 
+    right_hand_side = Field{Center, Center, Nothing}(grid)
+
     # Initialize vertically integrated lateral face areas
     ∫ᶻ_Axᶠᶜᶜ = Field{Face, Center, Nothing}(with_halo((3, 3, 1), grid))
     ∫ᶻ_Ayᶜᶠᶜ = Field{Center, Face, Nothing}(with_halo((3, 3, 1), grid))
```

Comment:
@simone-silvestri is 3 halos enough for this here?

Reply:
yep, actually one is enough for everything except for the `DiagonallyDominantInversePreconditioner` which does not work anyway at the moment